src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorChangeServiceRequest.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
+        CaseDetails<CaseData, State> details,
+        final CaseDetails<CaseData, State> beforeDetails
+    ) {
+        log.info("Solicitor change service request about to submit callback invoked with Case Id: {}", details.getId());
+
+        CaseData caseData = details.getData();
+        final Application application = caseData.getApplication();
+        final boolean isIssued = application.getIssueDate() != null;
+
+        if (application.isPersonalServiceMethod()) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(caseData)
+                .errors(singletonList("You may not select Personal Service. Please select Solicitor or Court Service."))
+                .build();
+        }
+
+        if (isIssued) {
+            if (application.isSolicitorServiceMethod()) {
+                log.info("Regenerate NOP for App and Respondent for case id: {}", details.getId());
+                details = caseTasks(generateApplicant1NoticeOfProceeding,
+                    generateApplicant2NoticeOfProceedings).run(details);
+            } else {
+                log.info("Regenerate NOP for App and Respondent, D10, and D84 for case id: {}", details.getId());
+                details = caseTasks(generateApplicant1NoticeOfProceeding,
+                    generateApplicant2NoticeOfProceedings, generateD10Form, generateD84Form).run(details);
+            }
+        }
+
+        final State state = application.isCourtServiceMethod() ? AwaitingAos : AwaitingService;
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(caseData)
+            .state(state)
+            .build();
+    }
+
+    public SubmittedCallbackResponse submitted(final CaseDetails<CaseData, State> details,
+                                               final CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("Solicitor change service request submitted callback invoked for case id: {}", details.getId());
+        final Application application = details.getData().getApplication();
+        final boolean isIssued = application.getIssueDate() != null;
+
+        if (isIssued) {
+            if (application.isSolicitorServiceMethod()) {
+                final User user = idamService.retrieveSystemUpdateUserDetails();
+                final String serviceAuthorization = authTokenGenerator.generate();
+
+                log.info("Submitting system-issue-solicitor-service-pack event for case id: {}", details.getId());
+                ccdUpdateService.submitEvent(details, SYSTEM_ISSUE_SOLICITOR_SERVICE_PACK, user, serviceAuthorization);
+
+                log.info("Send Notification to Applicant 1 Solicitor for case id: {}", details.getId());
+                applicationIssuedNotification.sendToApplicant1Solicitor(details.getData(), details.getId());
```

Comment:
Only question left for me (sorry!)

we're calling the notification directly here. I feel we should be using the NotificationDIspatcher instead. As thats the common theme?